### DISCUSSION TITLE
Questions on how Runelite pulls/stores player data for events like quest completions, levels, rare drops, etc.

### DISCUSSION THREAD

**Discussion by Tom-Cox-1** [runelite/runelite#16609] [Q&A]
Hi! Sorry if this isn't the best forum to ask, feel free to point me in the right direction if so.

I'm starting a personal project to create a "Personal OSRS Timeline", one where it would track the datetime for every medium to large accomplishment a player achieves in game. I think it would be a really cool project to work on, but I'm not exactly sure where to start.

My coding background is mostly data manip with sql/ python and some json, so the runelite API is a little intimidating at the moment. Does anyone have some good tutorial videos on an intro to runelite's API, as well as anything specific on pulling data similar to how Runelite does for its [screenshot](https://github.com/runelite/runelite/wiki/Screenshot) system?

Many thanks in advance!

**Comment by geheur** [comment]
There's not any official guides other than "copy how other plugins do it" and the build guide on the github wiki and the plugin hub guide in the plugin-hub repo README, and asking in the discord server.

There isn't a one-size-fits-all for the data you want. A lot stuff is stored in varbits; stuff like levels are not. And if you want to make a plugin like this that can pick up changes that happen while someone was playing on mobile, you probably need a different system than what screenshot uses to detect things.

The wikisync plugin might be worth looking at also in addition to the screenshots plugin.

**Reply by Tom-Cox-1** [reply]
Thanks for the info! Also good callout with the wikisync, I'll check there as well. First iteration most likely won't support mobile just to keep the scope small enough, but I'll try to keep that in mind for choices along the way so it hopefully one day could.

I'll check out those resources and follow up in the discord if I have more Qs, thanks for the info!

**Comment by discomethod** [comment]
If your interest in this project is to work on a fun coding problem then this suggestion is less helpful, but I personally use the Wise Old Man 3rd party stat tracker for OSRS which has existing frameworks for snapshotting levels/KCs and has an achievement system. They are also open source on Github. Perhaps iterating on that tool might give you a boost towards building the feature you're looking for.
